packit_service/worker/jobs.py

```diff
@@ -99,6 +99,13 @@ def get_handlers_for_event(
 
     matching_handlers: Set[Type["JobHandler"]] = set()
     for job in jobs_matching_trigger:
+        if (
+            isinstance(event, CheckRerunEvent)
+            and (event.job_identifier or job.identifier)
+            and event.job_identifier != job.identifier
```

Comment:
I think just the second condition should be enough.
The first condition only makes sure that they are not both empty/undefined, which they can't be if they are different (the second condition). But I'm not against having it explicitly stated.
```suggestion
            and event.job_identifier != job.identifier
```

Reply:
I wanted to be explicit, but it's not necessary as I am thinking about that now... I'll simplify that.